### ISSUE TITLE
Inefficient saving of results

### ISSUE THREAD

**Issue by franeknowak** [franeknowak/SwinCVS#6]
Currently script uses X.to('cpu'), which does not detach the values from the computation graph unnecessarily increasing resource use. This will be replaced with .cpu().detach() after validating to confirm it works as intended.
